File: frontend/src/service/CustomerService.js
import service from './api';

class CustomerService {

    search(search) {
        return service.post("/customers/search", search);
    }

    get(id) {
        return service.get(`/customers/${id}`);
    }

    create(data) {
        return service.post("/customers", data);
    }

    update(id, data) {
        return service.put(`/customers/${id}`, data);
    }

    delete(id) {
        return service.delete(`/customers/${id}`);
    }

    getClusteringCustomers() {
        return service.post("/customers/get-clustering-customers");
    }

    uploadFileExcel(objectFile) {
        return service.put(`/customers/upload-file`, objectFile, {
            headers: {
                'Content-type': 'multipart/form-data'
            }
        });
    }

    createCustomerData() {
        return service.post("/customers/create-customer-data");
    }

}

export default new CustomerService();

File: back-end/vrp-preprocess/src/main/java/com/vrp/demo/service/CustomerService.java
package com.vrp.demo.service;

import com.vrp.demo.entity.tenant.Customer;
import com.vrp.demo.exception.CustomException;
import com.vrp.demo.models.CustomerModel;
import com.vrp.demo.models.search.CustomerSearch;
import org.springframework.data.domain.Page;

import java.util.List;

public interface CustomerService extends BaseService<Customer, Long>{

    public List<CustomerModel> find(CustomerSearch search);

    public Page<CustomerModel> search(CustomerSearch search);

    public CustomerModel create(CustomerModel customerModel);

    public CustomerModel update(CustomerModel customerModel) throws CustomException;

    public int delete(Long id) throws CustomException;

    public CustomerModel findOne(Long id);

    public List<CustomerModel> getClusteringCustomers();

    public void createCustomerData();

    public List<CustomerModel> findAllWithCorrelations();

}
